packages/switch/src/LionButtonSwitch.js

```diff
@@ -0,0 +1,94 @@
+import { LionButton } from '@lion/button';
+import { html, css } from '@lion/core';
+
+export class LionButtonSwitch extends LionButton {
+  static get properties() {
+    return {
+      checked: {
+        type: Boolean,
+        reflect: true,
+      },
+    };
+  }
+
+  static get styles() {
+    return [
+      ...super.styles,
+      css`
+        :host {
+          display: inline-block;
+          position: relative;
+          width: 36px;
+          height: 16px;
+        }
+
+        .button-switch__track {
+          background: #eee;
+          width: 100%;
+          height: 100%;
+        }
+
+        .button-switch__thumb {
+          background: #ccc;
+          width: 50%;
+          height: 100%;
+          position: absolute;
+          top: 0;
+        }
+
+        :host([checked]) .button-switch__thumb {
+          right: 0;
+        }
+      `,
+    ];
+  }
+
+  render() {
+    return html`
+      <div class="button-switch__track"></div>
+      <div class="button-switch__thumb"></div>
+    `;
+  }
+
+  constructor() {
+    super();
+    this.role = 'switch';
+    this.checked = false;
+    this.addEventListener('click', this.__handleToggleStateChange);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute('aria-checked', `${this.checked}`);
+  }
+
+  firstUpdated(changedProperties) {
+    super.firstUpdated(changedProperties);
+    this.removeAttribute('type');
```

Comment:
Can't we do this in constructor?

Reply:
Also, this sets the type to null which is also not intended I think, this is an invalid value for type attribute.

I think just setting to type = button is fine right?